src/js/actions/layers.js

```diff
@@ -1587,6 +1590,155 @@ define(function (require, exports) {
+        var appStore = this.flux.store("application");
+
+        /**
+         * For search options: Get the layer type and if it is linked or an artboard, as an array of strings.
+         * This is used for comparison against the inputted search terms and any search filters.
+         * Not directly user-visible, but needs to be able to match user input.
+         *
+         * @private
+         * @param {Layer} layer
+         * @return {Array.<string>}
+        */
+        var _getLayerCategory = function (layer) {
+            var CATEGORIES = strings.SEARCH.CATEGORIES,
+                layerType = [CATEGORIES.LAYER];
+
+            if (layer.kind === layer.layerKinds.GROUP && layer.isArtboard) {
+                layerType.push(CATEGORIES.ARTBOARD);
+            } else {
+                // Find string associated with layer.kind, which is a number
+                _.forEach(Object.keys(layer.layerKinds), function (type) {
+                    if (layer.kind === layer.layerKinds[type]) {
+                        layerType.push(CATEGORIES[type].replace(" ", ""));
+                    }
+                });
+            }
+
+            if (layer.isLinked) {
+                layerType.push(CATEGORIES.LINKED);
+            }
+
+            return layerType;
+        };
+
+        /**
+         * For search options: get the layer ancestry
+         *
+         * @private
+         * @param {Layer} layer
+         * @return {string}
+        */
+        var _formatLayerAncestry = function (layer, appStore) {
+            var layerTree = appStore.getCurrentDocument().layers,
+                ancestors = layerTree.ancestors(layer),
+                ancestorNames = ancestors.first().name;
+                
+            return ancestors.skip(1).reduce(function (ancestors, ancestor) {
+                return ancestorNames += ("/" + ancestor.name);
+            }, ancestorNames);
+        };
+
+
+        /**
+         * Get list of layer options for search modules 
+         *
+         * @return {Immutable.List.<object>}
+        */
+        var getLayerSearchOptions = function () {
+            // Get list of layers
+            var document = appStore.getCurrentDocument();
+            if (!document) {
+                return {};
+            }
+            var layers = document.layers.allVisibleReversed,
+                layerMap = layers.map(function (layer) {
+                    var ancestry = _formatLayerAncestry(layer, appStore),
+                        layerType = _getLayerCategory(layer),
+                        iconID = svgUtil.getSVGClassFromLayer(layer);
+
+                    return {
+                        id: "layer_" + layer.id.toString(),
+                        title: layer.name,
+                        info: ancestry,
+                        displayInfo: ancestry,
+                        svgType: iconID,
+                        category: layerType,
+                        type: "item"
```

Comment:
I now see that with layer results, you are maintaining a really fine balance of keeping some information in search store, and some in the providers. Still, I think it would be really valuable to document, in search store, the exact structure of the item objects list you're expecting from the providers.